src/Controller/UserController.php

```diff
@@ -27,14 +33,33 @@ public function index(UserRepository $userRepository): Response
 
     /**
      * @Route("/new", name="user_new", methods="GET|POST")
+     * @param Request $request
+     * @param UserPasswordEncoderInterface $encoder
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
      */
-    public function new(Request $request): Response
+    public function new(Request $request, UserPasswordEncoderInterface $encoder): Response
     {
         $user = new User();
-        $form = $this->createForm(UserType::class, $user);
+        $form = $this->get('form.factory')->createBuilder(FormType::class, $user)
+            ->add('email', TextType::class)
+            ->add('password', PasswordType::class)
+            ->add('roles', CollectionType::class, [
```

Comment:
ensuite pour les roles, à tester, mais je pense que si tu fais un choiceType en  ça doit etre suffisant

Reply:
Pour les rôles, j'avais déjà testé sans le CollectionType j'ai une erreur de type Array to String conversion, c'est pour ça que je suis passé par le CollectionType.